js/gestionPresupuesto.js

```diff
@@ -1,25 +1,143 @@
-// TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+// TODO: Crear las funciones, objetos y variables indicadas en el  enunciado
+let gasto1 = new CrearGasto("descr");
 
 // TODO: Variable global
+let presupuesto = 0;
+let gastos = new Array();
+let idGasto = 0;
 
+function actualizarPresupuesto(valor) {
 
-function actualizarPresupuesto() {
-    // TODO
+    if (typeof valor === 'number' && valor > 0) { // Comprueba si valor es un número y es positivo
+        presupuesto = valor;
+        return presupuesto;
+    }
+    else {
+        console.log("El valor debe ser un número y no negativo");
+        return -1;
+    }
 }
 
 function mostrarPresupuesto() {
-    // TODO
+    return "Tu presupuesto actual es de " + presupuesto + " €";
 }
 
-function CrearGasto() {
-    // TODO
+function CrearGasto(descripcion, valor, fecha = Date.now(), ...etiquetas) { // Valor por defecto en fecha, seria fecha actual.
+
+    this.descripcion = descripcion;
+
+    if (valor > 0) { // Si el valor es positivo lo almacenamos, y si no asignamos a valor 0.
+        this.valor = valor;
+    } else {
+        this.valor = 0;
+    }
+
+    if (isNaN(Date.parse(fecha))) { // Si la fecha no tiene formato valido, almacenamos fecha actual.
+        this.fecha = Date.now();
+    }
+    else {
+        this.fecha = Date.parse(fecha); // Si fecha tiene formato valido, parseamos.
+    }
+
+    if (etiquetas === undefined) { // Si no se pasa el parametro de etiquetas, se crea un array vacío.
+        this.etiquetas = new Array();
+    }
+    else {
+        this.etiquetas = etiquetas;
```

Comment:
En lugar de pasar directamente el listado de etiquetas, puedes utilizar `this.anyadirEtiquetas` para eliminar duplicados, por ejemplo.